### ISSUE TITLE
Permission Denied issue while running Integration Automation (for e.g 200.xx Module)

### ISSUE THREAD

**Issue by gowdhaman-j** [cloud-native-toolkit/terraform-util-clis#78]
│ Error: External Program Execution Failed
│ 
│   with module.gitops-console-link-job.module.setup_clis.data.external.setup-binaries,
│   on .terraform/modules/gitops-console-link-job.setup_clis/main.tf line 22, in data "external" "setup-binaries":
│   22:   program = ["bash", "${path.module}/scripts/setup-binaries.sh"]
│ 
│ The data source received an unexpected error while attempting to execute the program.
│ 
│ Program: /bin/bash
│ Error Message: ln: /jq: Permission denied
│ ln: /igc: Permission denied
│ igc release not found
│ 
│ State: exit status 1

**Comment by timroster** [comment]
I see very similar errors to this trying to run tests for a module locally before on macos before pushing to a branch. It's in one of my modules, but I suspect it could be reproduced when locally running the tests for the vsi modules since that repo is highly derived from it.

```
╷
│ Error: External Program Execution Failed
│ 
│   with module.resource_group.module.clis.data.external.setup-binaries,
│   on .terraform/modules/resource_group.clis/main.tf line 22, in data "external" "setup-binaries":
│   22:   program = ["bash", "${path.module}/scripts/setup-binaries.sh"]
│ 
│ The data source received an unexpected error while attempting to execute the program.
│ 
│ Program: /bin/bash
│ Error Message: ln: /jq: Read-only file system
│ ln: /igc: Read-only file system
│ Error downloading yq3 from https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
│ 
│ State: exit status 1
╵
```

**Comment by timroster** [comment]
In my system - output in the `clis-debug.log`:

```
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (all) Input: {"bin_dir":"/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/bin2","clis":"yq,jq,igc","uuid":"pbgueilcrauqgcpb"}
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (all) Detected os type: macos
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (all) Detected architecture: amd64
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (jq) CLI already available in PATH. Creating symlink in
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (jq) CLI already available in PATH. Creating symlink in bin_dir
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (igc) CLI already available in PATH. Creating symlink in
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (igc) CLI already available in PATH. Creating symlink in bin_dir
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (yq3) Downloading cli: https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
/Users/timro/Public/ISV/terraform-vsi-zerotier-edge/test/stages/.terraform/modules/resource_group.clis/scripts: (yq3) Installing the cli in bin_dir                                                                                                                                               
```

Observations - the detected OS type is `macos` - yet the yq3 install logic tries linux. Need to investigate for bug in script logic. There are two link commands, the first to no path, the second to `bin_dir` - this explains the errors in the link commands. Probably could be ignored and the module would succeed so the root issue looks like the incorrect binary.

**Comment by timroster** [comment]
Observations above look to have a different potential root cause. The logic that is failing in `setup-yq3.sh` has nothing to do with potential issues in the `setup-igc.sh` in the original post. However the failed link attempts do match. Will move this to a new issue and work on PR for selection logic.

**Comment by timroster** [comment]
Both `setup-jq.sh` and `setup-igc.sh` make an early call to `setup-existing.sh`,  but this script lacks the logic present in `setup-binary.sh` to determine a value for `${BIN_DIR}` - causing https://github.com/cloud-native-toolkit/terraform-util-clis/blob/cbd9be5b710a40d9f0c6388245b5777cdbda6016/scripts/setup-existing.sh#L26 to execute as:

```
ln -s ${COMMAND} "/${CLI_NAME}"
```

This is causing the permission denied to appear in the terraform log. These are not fatal to the terraform execution.

**Comment by timroster** [comment]
further testing seems by @gowdhaman-j is now clearing after #81 . Marking as resolved and closing
